### PR TITLE
[FIX BUG] text_splitter.py

### PR DESCRIPTION
**Pull request by jrredondo**
Fixed code when buffer_size is greater than 1

### CODE REVIEW

File: libs/experimental/langchain_experimental/text_splitter.py [hunk 1]
```diff
@@ -238,12 +238,12 @@ def split_text(
             end_index = index
 
             # Slice the sentence_dicts from the current start index to the end index
-            group = sentences[start_index : end_index + 1]
+            group = sentences[start_index : end_index + self.buffer_size]
```
Comment (baskaryan) [inline libs/experimental/langchain_experimental/text_splitter.py:241]: why are we adding buffer_size here?

**Review by baskaryan — COMMENTED** [review]
not sure i understand the bug / fix, could you add a test case to demonstrate?

File: libs/experimental/langchain_experimental/text_splitter.py (hunk 1 above)
Comment (jrredondo) [reply libs/experimental/langchain_experimental/text_splitter.py:241]: To answer this question I have another one ;-) Why are we adding +1 here?

Suppose this example:

```
text_splitter = CustomSemanticChunker(embeddings=embeddings, breakpoint_threshold_type="percentile", breakpoint_threshold_amount=70, buffer_size=1) 

chunks=text_splitter.split_text("sentence1. sentence2. sentence3. sentence4. sentence5. sentence6. sentence7. sentence8. sentence9. sentence10.")
```

Combined sentences in this case are (buffer_size=1):

```
{'sentence': 'sentence1.', 'index': 0, 'combined_sentence': 'sentence1. sentence2.'} 
{'sentence': 'sentence2.', 'index': 1, 'combined_sentence': 'sentence1. sentence2. sentence3.'}
{'sentence': 'sentence3.', 'index': 2, 'combined_sentence': 'sentence2. sentence3. sentence4.'}
{'sentence': 'sentence4.', 'index': 3, 'combined_sentence': 'sentence3. sentence4. sentence5.'}
{'sentence': 'sentence5.', 'index': 4, 'combined_sentence': 'sentence4. sentence5. sentence6.'}
{'sentence': 'sentence6.', 'index': 5, 'combined_sentence': 'sentence5. sentence6. sentence7.'}
{'sentence': 'sentence7.', 'index': 6, 'combined_sentence': 'sentence6. sentence7. sentence8.'}
{'sentence': 'sentence8.', 'index': 7, 'combined_sentence': 'sentence7. sentence8. sentence9.'}
{'sentence': 'sentence9.', 'index': 8, 'combined_sentence': 'sentence8. sentence9. sentence10.'}
{'sentence': 'sentence10.', 'index': 9, 'combined_sentence': 'sentence9. sentence10.'}
```
And distances: [0.0574508136828753, 0.0592057580219334, 0.06952031157395344, 0.07423465529901396, 0.07651917783119067, 0.08379923453678395, 0.07362624904547133, 0.07092059885224533, 0.03802994498710943]

With breakpoints at: **[3, 4, 5]** and distance threshold of **0.0739912927975969**

So we have this chunks:
**sentence1. sentence2. sentence3. sentence4.
sentence5.
sentence6.
sentence7. sentence8. sentence9. sentence10.**

According to this, the first breakpoint is due to a 0.074 distance between combined sentence corresponding to buffers of sentence 3 an 4:

{'sentence': 'sentence3.', 'index': 2, 'combined_sentence': 'sentence2. sentence3. sentence4.'} 
{'sentence': 'sentence4.', 'index': 3, 'combined_sentence': 'sentence3. sentence4. sentence5.'} 

But, if the breakpoints are between sentences 3 and 4, why we obtain those fragments?
In this case, the logical answer should be:

**sentence1. sentence2. sentence3.
sentence4.
sentence5.
sentence6. sentence7. sentence8. sentence9. sentence10.**

Removing the +1.

Adding the +1 we are considering a  hardcoded buffer size of +1 in my opinion when you do the splits.
So in my honest opinion I would remove +1 or add +self.buffer_size, as you prefer.

I hope I have explained myself well. Please, let me know if I am wrong ;-)
Best regards.

**Comment by efriis** [comment]
closing as unclear / test request not acted upon. In the future concise descriptions are better! Please read our review process for more recommendations. https://python.langchain.com/docs/contributing/review_process/

**Pull request closed without merging**